segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -759,15 +763,20 @@ private SystemSnapshotRecord readSnapshotRecord(SnapshotInfo snapshotInfo, byte[
                     Preconditions.checkState(!segmentSnapshot.segmentMetadata.getFirstChunk().equals(segmentSnapshot.segmentMetadata.getLastChunk()),
                             "First chunk and last chunk should not match. Segment snapshot= %s", segmentSnapshot);
                 }
+                int dataSize = 0;
                 ChunkMetadata previous = null;
                 for (val metadata : segmentSnapshot.getChunkMetadataCollection()) {
+                    dataSize += metadata.getLength();
```

Comment:
Any chance this might exceed the bounds for Integer?

Reply:
Thanks. Good catch ! 
metadata.length is long so I don't understand why there was no compiler warning in the first place.